dsi/plugins/env.py

```diff
@@ -27,13 +27,17 @@ def __init__(self, path=None) -> None:
         self.posix_info['gid_list'] = os.getgrouplist(moniker, egid)
         # Plugin output collector
         self.output_collector = OrderedDict()
+        # class bool to track whether header has been written.
+        self.header_exist = False
```

Comment:
If we want to pack the header exactly once before add_row functionality happens, why not do it in `__init__` and avoid having this flag? column_cnt could also definitely be set after initialization instead of it being None until you add a row.

Reply:
I want Plugin implementations to be as lazy as possible before sampling in the structured schema, because once the schema is defined we can only add samples of that schema. The expectation is that less trivial Plugin implementations will implement methods beyond the required abstract methods that add or remove things from the output_collector based on internal logic.

To increase laziness, I only call pack_header when sampling begins (with the first add_row). Although checking the bool for each add_row takes time, it allows me to totally ignore schema until the last possible moment.

Curious for your thoughts on this. We could:
1. Improve documentation to make my wish for laziness clear both in the source and in the Sphinx docs.
2. Change my mind about the lazy approach.

Reply:
Yeah I can buy that, maybe some plugins require significant work before the schema is known and they shouldn't have to shove all of that logic in init. Makes sense.

Reply:
Does this help? https://github.com/lanl/dsi/pull/23/commits/4b427a9f2e041a5f9755cec2b4c5f39c75e2e1cd 

If so hit resolve, if not, please edit and I'll check it out.